inc/class-shortcode-ui.php

```diff
@@ -191,23 +190,33 @@ private function compare_shortcodes_by_label( $a, $b ) {
 	public function handle_ajax_do_shortcode() {
 
 		// Don't sanitize shortcodes — can contain HTML kses doesn't allow (e.g. sourcecode shortcode)
-		$shortcode = ! empty( $_POST['shortcode'] ) ? stripslashes( $_POST['shortcode'] ) : null;
-		$post_id   = ! empty( $_POST['post_id'] ) ? intval( $_POST['post_id'] ) : null;
+		if ( ! empty( $_POST['shortcode'] ) ) {
+			$shortcode = stripslashes( $_POST['shortcode'] ); // WPCS: input var okay
+		} else {
+			$shortcode = null;
+		}
+		if ( isset( $_POST['post_id'] ) ) {
+			$post_id = intval( $_POST['post_id'] ); // WPCS: input var okay
+		} else {
+			$post_id = null;
+		}
 
-		if ( ! current_user_can( 'edit_post', $post_id ) || ! wp_verify_nonce( $_POST['nonce'], 'shortcode-ui-preview' ) ) {
+		if ( ! current_user_can( 'edit_post', $post_id ) || ! wp_verify_nonce( $_POST['nonce'], 'shortcode-ui-preview' ) ) { // WPCS: input var okay
```

Comment:
`$_POST['nonce']` is getting flagged as not validated because there is no `isset()` or `empty()` check. E.g.: `isset( $_POST['nonce'] )`. It isn't considered sanitized because `wp_verify_nonce()` isn't a sanitization function (see WordPress-Coding-Standards/WordPress-Coding-Standards#282)